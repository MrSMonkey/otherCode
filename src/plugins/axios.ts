/*
 * @Description: axios网络请求库二次封装
 * @Author: LiuZhen
 * @Date: 2018-09-18 11:49:38
 * @Last Modified by: linyu
 * @Last Modified time: 2019-04-09 15:29:08
 */
import axios from 'axios';
import store from '../store';
import router from '../router';
import Vue from 'vue';
const Axios = axios.create({
  timeout: 500000,
  headers: {
    'X-Requested-With': 'XMLHttpRequest'
  }
});

// http request请求拦截器(所有请求发送都要执行的操作)
Axios.interceptors.request.use(
  (config: any) => {
    // 根据环境设置baseURL
    if (process.env.NODE_ENV === 'production' && process.env.VUE_APP_TITLE === 'production') {  // 生产
      config.baseURL = 'https://api-gateway.uoko.com/';
    } else if (process.env.NODE_ENV === 'production' && process.env.VUE_APP_TITLE === 'test') {  // 测试
      config.baseURL = 'http://api-gateway.testuoko.com/';
    } else if (process.env.NODE_ENV === 'production' && process.env.VUE_APP_TITLE === 'pre-release') {  // pre
      config.baseURL = 'https://api-gateway-pre.uoko.com/';
    } else {
      // config.baseURL = 'http://front-end.testuoko.com:3000/mock/22/'; // mock地址
      // config.baseURL = 'http://192.168.200.44:7070/';
      // config.baseURL = 'http://api-gateway.testuoko.com/';
      config.baseURL = 'http://172.16.3.103:8008';
    }

    /*登录授权, 登录接口修改 Authorization */
    if (config.url.indexOf('/auth/asset/register_login/web/mobile') > -1
      || config.url.indexOf('/verification_code') > -1) {
      config.headers.Authorization = 'Basic b3duZXI6MTIzNDU2';
    } else {
      const token: string | null = localStorage.getItem('siteToken');
      config.headers.Authorization = token !== null ? `Bearer ${ token}` : '';
    }

    return config;
  }, (error) => {
    return Promise.reject(error);
  }
);

// http response响应拦截器
Axios.interceptors.response.use(
  (response: any) => {
    // 这里可以做一些响应拦截的操作
    if (response.status === '401' || response.data.code === '70001' || response.data.code === '20001') {
      localStorage.removeItem('siteToken'); // 清除token
      localStorage.removeItem('userId'); // 清除userId
      router.push(`/bind?redirectUrl=${window.location.href.split('#')[1]}`);
    }
    return response.data;
  }, (error) => {
    // const config: any = error.config;
    const status: any = error.response.status || 500;
    if (status === 401) {
      Vue.prototype.$toast({
        duration: 3000,       // 持续展示 toast
        type: 'fail',
        message: `登录失效`
      });
      setTimeout(() => {
        router.push({
          path: '/bind'
        });
      }, 3000);
    } else {
      Vue.prototype.$toast({
        duration: 3000,       // 持续展示 toast
        type: 'fail',
        message: `服务器异常`
      });
    }
    // return Promise.reject(error);
  }
);

export default Axios;
