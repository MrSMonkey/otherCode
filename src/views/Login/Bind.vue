/*
 * @Description: 登陆页面
 * @Author: chenmo
 * @Date: 2019-02-15 13:42:19
 * @Last Modified by: chenmo
 * @Last Modified time: 2019-02-20 15:58:52
 */
<template>
  <section class="login">
    <h1>登录</h1>
    <section class="el-input">
      <p class="tip">手机号<span>(*新用户初次登录将自动注册)</span></p>
      <div style="marginBottom: 20px" class="phone">
        <van-field
          v-model="phone"
          placeholder="请输入手机号"
          type="text"
          clearable
        />
        <p class="error-info" ref="phoneErrorInfo"></p>
      </div>
      <p class="tip">验证码</p>
      <section>
        <section class="code-box">
          <div class="code">
            <van-field
              v-model="code"
              type="text"
              placeholder="验证码"
              clearable
            />
            <p class="error-info" ref="codeErrorInfo"></p>
          </div>
          <div>
            <van-button slot="button" size="normal" type="default" class="btn" v-if="!isphoneErr">获取验证码</van-button>
            <span v-else>
              <van-button slot="button" size="normal" type="default" class="btn" v-if="isDisabledVerfiyCodeBtn">重发({{time}})</van-button>
              <van-button slot="button" size="normal" type="default" class="btn active" v-else @click="getCode">获取验证码</van-button>
            </span>
          </div>
        </section>
        
      </section>
      <van-button slot="button" size="large" type="default" class="login-btn" v-if="!isphoneErr || !isCodeErr">登录</van-button>
      <van-button slot="button" size="large" type="default" class="login-btn-active" v-else @click="submitLogin" loading-text="登录中..." :loading="loading">登录</van-button>
    </section>
  </section>
</template>

<script lang="ts">
import { Component, Vue, Watch } from 'vue-property-decorator';
import { State, Getter, Mutation, Action } from 'vuex-class';
import CommonMixins from '@/utils/mixins/commonMixins';
import { handleWebStorage } from '@/utils/utils';
import { Field, Row, Col, Button } from 'vant';
import api from '@/api';
// global vuex
const namespace: string = 'global';
// 声明引入的组件
@Component({
  name: 'Bind',
  components: {
    [Field.name]: Field,
    [Row.name]: Row,
    [Col.name]: Col,
    [Button.name]: Button,
  }
})
// 类方式声明当前组件
export default class Bind extends CommonMixins {
   public $refs!: {
    [key: string]: any,
    phoneErrorInfo: HTMLFormElement,
  };

  @Mutation('updateToken', { namespace }) private updateToken: any;

  private phone: string = '';
  private code: string = '';
  private isphoneErr: boolean = false; // 校验手机号
  private isCodeErr: boolean = false; // 校验验证码
  private loading: boolean = false; // 登陆按钮loading
  private time: number = 60;
  private isDisabledVerfiyCodeBtn: boolean = false; // 验证码是否发送

  /**
   * @description 获取验证码
   * @returns void
   * @author chenmo
   */
  private async getCode() {
    try {
      const res: any = await this.axios.post(api.getCode + `/${this.phone}`);
      if (res && res.code === '000') {
        this.startCountdown(); // 开始倒计时
      } else {
        this.$toast('获取验证码失败');
      }
    } catch (err) {
      throw new Error(err || 'Unknow Error!');
    }
  }
  /**
   * @description // 显示倒计时
   * @params phone 手机号
   * @return void
   * @author chenmo
   */
  private startCountdown() {
    this.isDisabledVerfiyCodeBtn = true;
    const timer = setInterval(() => {
      this.time--;
      if (this.time === 0) {
        this.isDisabledVerfiyCodeBtn = false;
        this.time = 60;
        clearInterval(timer);
      }
    }, 1000);
  }

  /**
   * @description 登录
   * @return void
   * @author chenmo
   */
  private async submitLogin() {
    try {
      const res: any = await this.axios.post(api.login, {
        mobile: this.phone,
        verificationCode: this.code,
        registerSource: 1
      });
      this.loading = true;
      if (res && res.code === '000') {
        handleWebStorage.setLocalData('siteToken', res.data.access_token); // 本地存储token
        handleWebStorage.setLocalData('userId', res.data.userId); // 本地存储userId
        this.updateToken(res.data.access_token);
        this.$router.push('/house'); // 跳转到房源列表
      } else {
        this.$toast(res.msg || '登录失败');
      }
    } catch (err) {
      throw new Error(err || 'Unknow Error!');
    } finally {
      this.loading = false;
    }
  }
  // Watch
  @Watch('phone')
  private handlerPhone(newVal: string) {
    if (newVal && /^1[345789]\d{9}$/.test(newVal)) {
      this.$refs.phoneErrorInfo.innerHTML = '';
      this.isphoneErr = true;
    } else {
      this.isphoneErr = false;
      this.$refs.phoneErrorInfo.innerHTML = '请输入正确的手机号';
    }
  }

  @Watch('code')
  private handlerCode(newVal: string) {
    if (newVal && /^\d{6}$/ .test(newVal)) {
      this.$refs.codeErrorInfo.innerHTML = '';
      this.isCodeErr = true;
    } else {
      this.isCodeErr = false;
      this.$refs.codeErrorInfo.innerHTML = '请输入6位手机验证码';
    }
  }
}
</script>

<style lang="stylus" rel="stylesheet/stylus">
@import '../../assets/stylus/main.styl'
.login
  padding vw(20)
  min-height 100%
  background $global-background
  h1
    font-size 34px
    color $next-text-color
    margin-bottom vw(20)
  .el-input
    .tip
      font-family PingFang-SC-Medium
      font-size 12px
      color $tip-text-color
      margin-bottom 6px
      span
        color $main-color
    .error-info
      color $warning-color
      font-size 12px
      margin-top vw(6)
    .phone
      flex 1
      -webkit-appearance none
      border none
      outline none
      height 56px
      font-size 14px /* no */
      color #474747
      background #f8f8f8
      border 1px solid #f8f8f8
      padding-top vw(5) 
      width 100%
      border-radius 4px
      .van-field 
        background #f8f8f8
        &::after
          border-bottom: 0 !important
    .code-box
      display -webkit-flex
      display flex
      justify-content space-between
      .code
        flex 1
        -webkit-appearance none
        border none
        outline none
        height 56px
        font-size 14px /* no */
        color #474747
        background #f8f8f8
        border 1px solid #f8f8f8
        padding-top vw(5) 
        border-radius 4px
        margin-right vw(20)
        width 50%
        .van-field 
          background #f8f8f8
          display inline-block
          &::after
            border-bottom: 0 !important
      .btn 
        height 56px
        line-height 56px
        color #fff
        background-color rgb(216, 216, 216)
        border-radius 4px
        padding 0 30px
      .active
        background $main-color
    .login-btn
      color #fff
      background rgb(216, 216, 216)
      border-radius 4px
      margin-top vw(20)
    .login-btn-active
      color #fff
      background $main-color
      border-radius 4px
      margin-top vw(20)
</style>
