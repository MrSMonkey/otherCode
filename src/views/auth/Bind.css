.header {
  margin: 26px 0 36px 30px;
  font-size: 34px; /* no */
  color: #757575;
  font-weight: bold;
}
.view {
  background: #fff;
  min-height: 100%;
  overflow: hidden;
}
.main {
  padding: 0 30px;
}
.section {
  margin-bottom: 50px;
}
.section-tit {
  font-size: 12px; /* no */
  color: #b8b8b8;
  margin-bottom: 10px;
  span {
    color: #c3aa82;
  }
}

.field {
  padding: 30px 20px;
  border-radius: 8px;
  background: #f8f8f8;
  border: 1px solid #f8f8f8; /* no */
  display: flex;
  padding-right: 0;
  box-sizing: border-box;
  position: relative;
  align-items: center;
  &.has-error {
    background: rgba(255, 82, 82, 0.1);
    border-color: #ff5252;
  }

  &.disabled {
    background: #f2f3f5;
    border-color: #f2f3f5;
  }
}
.input {
  flex: 1;
  -webkit-appearance: none;
  border: none;
  outline: none;
  height: 36px;
  line-height: 36px;
  font-size: 14px; /* no */
  color: #474747;
  padding: 0;
  background: transparent;
  padding: 0 24px;
  width: 100%;
}
.clear-btn {
  padding: 0 30px;
  &:active {
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0.5);
  }
}

.error-msg {
  position: absolute;
  left: 0;
  font-size: 12px; /* no */
  color: #ff5252;
  top: 100%;
}

.select {
  position: relative;
  padding-right: 26px;
  border-right: 1px solid #eaeaea; /* no */
}
.select-val {
  margin-right: 25px;
}
.select-hidden {
  opacity: 0;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
}

.validate-field {
}
.field-wrap {
  width: 420px;
  margin-right: 20px;
  display: inline-block;
}
.getcode-btn {
  display: inline-block;
  height: 106px;
  line-height: 106px;
  width: 250px;
  text-align: center;
  background: #c3aa82;
  border-radius: 8px;
  color: #fff;
  vertical-align: middle;

  &:active {
    background: #ab8850;
    transition: background 0.2s ease;
  }

  &.disabled {
    background: #d8d8d8;
  }
}

.login-btn {
  width: 100%;
  display: block;
  height: 98px;
  border-radius: 8px;
  color: #fff;
  background: #c3aa82;
  text-align: center;
  line-height: 98px;
  margin-top: 80px;
  font-size: 16px; /* no */
  &.disabled {
    background: #d8d8d8;
  }
  &:active {
    background: #ab8850;
    transition: background 0.2s ease;
  }
}

.go-reg {
  margin: 0 30px;
  margin-top: 60px;
  position: relative;
  text-align: center;
  &:after {
    content: '';
    position: absolute;
    left: 0;
    top: 50%;
    width: 100%;
    height: 1px; /* no */
    background: #d8d8d8;
  }
}
.go-reg-txt {
  display: inline-block;
  background: #fff;
  position: relative;
  z-index: 1;
  padding: 0 35px;
}
