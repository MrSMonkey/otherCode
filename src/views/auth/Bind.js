import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import CSSModules from 'react-css-modules'
import { Link, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import Icon from '../../components/Icon'
import verify from '../../utils/verify'
import Button from '@/components/Button'
import Loading from 'react-loading'
import { actions as bindActions } from '../../reducers/bind'
import { actions as appActions } from '../../reducers/app'
import { px2vw, vw2px, parseQuery } from '../../utils'
import _ from 'lodash'
import styles from './Bind.css'

const ReactLoading = () => {
  const loadingTheme = {
    width: vw2px(px2vw(30)),
    height: vw2px(px2vw(30)),
    color: 'rgba(0,0,0,0.8)',
    type: 'spokes'
  }

  return <Loading {...loadingTheme} />
}

/**输入框 */
@CSSModules(styles, { allowMultiple: true })
class InputField extends PureComponent {
  static propTypes = {
    addonBefore: PropTypes.func,
    disabled: PropTypes.bool,
    error: PropTypes.string,
    value: PropTypes.string,
    onInput: PropTypes.func,
    onReset: PropTypes.func,
    isLoading: PropTypes.bool
  }

  static defaultProps = {
    addonBefore: () => null,
    disabled: false,
    error: '',
    value: '',
    onInput: () => null,
    onReset: () => null,
    onBlur: () => null,
    isLoading: false
  }

  render() {
    const {
      addonBefore,
      disabled,
      error,
      value,
      onInput,
      onReset,
      isLoading,
      onBlur,
      ...other
    } = this.props
    return (
      <div
        styleName={`field ${error ? 'has-error' : ''} ${
          disabled ? 'disabled' : ''
        }`}
      >
        {addonBefore(this.props)}
        <input
          onChange={onInput}
          onBlur={onBlur}
          value={value}
          styleName="input"
          {...other}
        />
        <a styleName="clear-btn" onClick={onReset}>
          {isLoading ? (
            <ReactLoading type="bubbles" />
          ) : (
            <Icon name="clear_x" />
          )}
        </a>
        {error && <p styleName="error-msg">{error}</p>}
      </div>
    )
  }
}

@CSSModules(styles)
class ThemedSelect extends PureComponent {
  static defaultProps = {
    onSelect: () => null,
    options: []
  }
  static propTypes = {
    onSelect: PropTypes.func,
    options: PropTypes.array
  }

  state = {
    value:
      this.props.options[0] === undefined ? '' : this.props.options[0].value
  }

  onSelect = e => {
    const v = e.target.value
    this.setState({
      value: v
    })
    this.props.onSelect(v)
  }

  render() {
    const { value } = this.state
    /*   const { options } = this.props */

    return (
      <div styleName="select">
        <span styleName="select-val">{value}</span>
        {/*    <Icon name="caret" />
        <select
          styleName="select-hidden"
          value={value}
          onChange={this.onSelect}
        >
          {options.map(o => (
            <option key={o.value} value={o.value}>
              {o.label}
            </option>
          ))}
        </select> */}
      </div>
    )
  }
}

/**获取验证码按钮 */
@CSSModules(styles, { allowMultiple: true })
class ValidationCodeButton extends PureComponent {
  static propTypes = {
    onCountEnd: PropTypes.func,
    onClick: PropTypes.func
  }
  static defaultProps = {
    onCountEnd: () => null,
    onClick: () => null
  }
  timeoutId = null
  totalCount = 60
  state = {
    isCounting: false,
    count: this.totalCount
  }
  countDown = fn => {
    let count = this.state.count
    if (!count) {
      fn && fn()
      return
    }
    this.timeoutId = setTimeout(() => {
      count--
      this.setState({
        count
      })
      this.countDown(fn)
    }, 1000)
  }
  countEnd = () => {
    this.countReset()
    this.props.onCountEnd()
  }
  countStart = () => {
    this.setState({
      isCounting: true,
    })
    this.countDown(this.countEnd)
  }
  countReset = () => {
    clearTimeout(this.timeoutId)
    this.setState({
      count: this.totalCount,
      isCounting: false
    })
  }
  onCodeButtonClick = () => {
    if (this.state.isCounting || this.state.isDisabled) { return }

    this.countStart()
    this.props.onClick()
  }
  
  componentWillUnmount() {
    this.countReset()
  }

  render() {
    const { isDisabled } = this.props
    const { count, isCounting } = this.state
    return (
      <div styleName="getcode-btn">
        <Button type="Primary" disabled={isDisabled || isCounting} radius onClick={this.onCodeButtonClick}>
          <span>{isCounting ? `重发 (${count})` : '获取验证码'}</span>
        </Button>
      </div>
    )
  }
}

@CSSModules(styles, { allowMultiple: true })
class Bind extends PureComponent {
  state = {
    phoneError: '',
    isPhoneError: true,
    codeError: '',
    isCodeError: true,
  }
  validatePhone = phone => {
    const { success } = verify.phone(phone)

    this.setState({
      phoneError: success ? '' : '请输入正确的手机号码',
      isPhoneError: !success
    })
    return success
  }
  onPhoneBlur = () => {
    const { phone } = this.props
    const check = this.validatePhone(phone)

    if (!check) {
      return check
    }
  }
  validatePhoneField = () => {
    const { phone, isCheckingPhone, checkError } = this.props
    const check = this.validatePhone(phone)

    return check && (!isCheckingPhone && !checkError)
  }
  validateCode = identifyingCode => {
    identifyingCode = _.trim(identifyingCode)
    this.setState({
      codeError: identifyingCode ? '' : '请输入验证码',
      isCodeError: !identifyingCode
    })

    return identifyingCode
  }
  onCodeBlur = () => {
    const { identifyingCode } = this.props
    this.validateCode(identifyingCode)
  }

  onBindClick = () => {
    // const pv = this.validatePhoneField()
    const cv = this.validateCode(this.props.identifyingCode)
    if (cv) {
      this.props.bindStart()
    }
  }

  render() {
    const {
      updatePhone,
      updateCode,
      genCode,
      isCheckingPhone,
      router,
      bindSuccess,
      phone,
      bindLoading,
      identifyingCode
    } = this.props
    const { phoneError, codeError, isPhoneError, isCodeError } = this.state
    const redirectUrl = parseQuery(router.location.search).redirect || '/parthner/houses'

    return (
      <div styleName="view">
        <h1 styleName="header">登录</h1>
        <div styleName="main">
          <div styleName="section">
            <div styleName="section-tit">
              手机号 (<span>*新用户初次登录将自动注册</span>)
            </div>
            <InputField
              placeholder="请输入手机号码"
              onBlur={this.onPhoneBlur}
              value={phone}
              onInput={e => updatePhone(e.target.value)}
              onReset={e => updatePhone('')}
              isLoading={isCheckingPhone}
              error={phoneError}
            />
          </div>

          <div styleName="section">
            <div styleName="section-tit">验证码</div>
            <div styleName="validate-field">
              <div styleName="field-wrap">
                <InputField
                  value={identifyingCode}
                  onBlur={this.onCodeBlur}
                  error={codeError}
                  onInput={e => updateCode(e.target.value)}
                  onReset={e => updateCode('')}
                  placeholder="请输入验证码"
                />
              </div>
              <ValidationCodeButton
                isDisabled={isPhoneError}
                validate={this.validatePhoneField}
                onClick={genCode}
              />
            </div>
          </div>

          <Button type="Primary" disabled={bindLoading || isPhoneError || isCodeError} radius onClick={this.onBindClick}>
            {bindLoading ? '登录中，请稍等...' : '登录'}
          </Button>
        </div>
        {bindSuccess ? <Redirect to={decodeURIComponent(redirectUrl)} /> : null}
      </div>
    )
  }
}

export default connect(
  state => ({
    ...state.bind,
    router: state.router
  }),
  {
    ...bindActions,
    getUserInfo: appActions.getUserInfo
  }
)(Bind)
