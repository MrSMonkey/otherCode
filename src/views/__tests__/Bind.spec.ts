/*
 * @Description: 底部按钮单元组件测试
 * @Author: chenmo
 * @Date: 2019-03-26 15:28:22
 * @Last Modified by: chenmo
 * @Last Modified time: 2019-03-27 11:51:02
 */

import Vue from 'vue';
import axios from 'axios';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Bind from '@/views/Bind.vue';

// 独立构建Vue构造函数
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

const mockData = {
  data: {
    answer: 'mock_yes',
    image: 'mock.png'
  }
};
jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve(mockData))
}));

describe('对Bind组件进行的单元测试', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        global: {}
      }
    });

    // 挂载组件
    wrapper = shallowMount(Bind, {
      store,
      localVue
    });

    // 来确保每次开始前都重置模块和mock依赖的状态
    jest.resetModules();
    jest.clearAllMocks();
  });

  afterEach(() => {
    // 销毁实例
    wrapper.destroy();
  });

  test('BottomBtn', () => {
    // 测试实例是Vue组件
    expect(wrapper.isVueInstance()).toBeTruthy();
    // 存在 BottomBtn 组件
    expect(wrapper.is(Bind)).toBeTruthy();
    // 验证组件是否渲染
    expect(wrapper.contains('.login')).toBeTruthy();
  });

  // // 点击按钮后调用了 getCode 方法
  // it('getCode Fn should be called', () => {
  //   const mockFn = jest.fn();
  //   wrapper.setMethods({getCode: mockFn});
  //   wrapper.find('.btn').trigger('click');
  //   expect(mockFn).toBeCalled();
  // });

  // // 可以用 Async/Await 测试 axios.get 方法返回值
  // it('Calls get promise result', async () => {
  //   const result = await wrapper.vm.getCode();
  //   expect(result).toEqual(mockData);
  // });
  // 生成测试快照
  test('create expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
