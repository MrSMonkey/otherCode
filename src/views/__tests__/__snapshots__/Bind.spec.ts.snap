// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`对Bind组件进行的单元测试 create expected html structure 1`] = `
<section
  class="login"
>
  <h1>
    登录
  </h1>
   
  <section
    class="el-input"
  >
    <p
      class="tip"
    >
      手机号
      <span>
        (*新用户初次登录将自动注册)
      </span>
    </p>
     
    <div
      class="phone"
      style="margin-bottom: 20px;"
    >
      <van-field-stub
        border="true"
        clearable="true"
        placeholder="请输入手机号"
        type="text"
        value=""
      />
       
      <p
        class="error-info"
      />
    </div>
     
    <p
      class="tip"
    >
      验证码
    </p>
     
    <section>
      <section
        class="code-box"
      >
        <div
          class="code"
        >
          <van-field-stub
            border="true"
            clearable="true"
            placeholder="验证码"
            type="text"
            value=""
          />
           
          <p
            class="error-info"
          />
        </div>
         
        <div>
          <van-button-stub
            class="btn"
            loadingsize="20px"
            size="normal"
            slot="button"
            tag="button"
            type="default"
          >
            获取验证码
          </van-button-stub>
        </div>
      </section>
    </section>
     
    <van-button-stub
      class="login-btn"
      loadingsize="20px"
      size="large"
      slot="button"
      tag="button"
      type="default"
    >
      登录
    </van-button-stub>
  </section>
</section>
`;
